rotkehlchen/chain/evm/decoding/balancer/v1/decoder.py

```diff
@@ -0,0 +1,392 @@
+                if (
+                    context.event.event_type == HistoryEventType.SPEND and
+                    context.event.event_subtype == HistoryEventSubType.NONE
+                ):
+                    context.event.event_subtype = HistoryEventSubType.RETURN_WRAPPED
+                    context.event.counterparty = CPT_BALANCER_V1
+                    context.event.notes = f'Return {context.event.balance.amount} {context.token.symbol} to a Balancer v1 pool'  # noqa: E501
+                    return TransferEnrichmentOutput(matched_counterparty=CPT_BALANCER_V1)
+
+        return FAILED_ENRICHMENT_OUTPUT
+
+    def _decode_non_proxy_events(self, context: DecoderContext) -> DecodingOutput:
+        """Not all balancer v1 pools are created via the UI. This method decodes the events of such pools."""  # noqa: E501
+        if context.tx_log.topics[0] == JOIN_V1:
+            user_address = bytes_to_address(context.tx_log.topics[1])
+            token_in = self.base.get_or_create_evm_token(address=bytes_to_address(context.tx_log.topics[2]))  # noqa: E501
+            deposit_amount = asset_normalized_value(
+                amount=int.from_bytes(context.tx_log.data[:32]),
+                asset=token_in,
+            )
+            return DecodingOutput(action_items=[ActionItem(
+                action='transform',
+                from_event_type=HistoryEventType.SPEND,
+                from_event_subtype=HistoryEventSubType.NONE,
+                to_event_type=HistoryEventType.DEPOSIT,
+                location_label=user_address,
+                to_event_subtype=HistoryEventSubType.DEPOSIT_ASSET,
+                to_counterparty=CPT_BALANCER_V1,
+                to_notes=f'Deposit {deposit_amount} {token_in.symbol} to a Balancer v1 pool',
+                asset=token_in,
+            )])
+
+        if context.tx_log.topics[0] == EXIT_V1:
+            user_address = bytes_to_address(context.tx_log.topics[1])
+            token_out = self.base.get_or_create_evm_token(address=bytes_to_address(context.tx_log.topics[2]))  # noqa: E501
+            withdrawn_amount = asset_normalized_value(
+                amount=int.from_bytes(context.tx_log.data[:32]),
+                asset=token_out,
+            )
+            return DecodingOutput(action_items=[ActionItem(
+                action='transform',
+                from_event_type=HistoryEventType.RECEIVE,
+                from_event_subtype=HistoryEventSubType.NONE,
+                to_event_type=HistoryEventType.WITHDRAWAL,
+                location_label=user_address,
+                to_event_subtype=HistoryEventSubType.REMOVE_ASSET,
+                to_counterparty=CPT_BALANCER_V1,
+                to_notes=f'Receive {withdrawn_amount} {token_out.symbol} after removing liquidity from a Balancer v1 pool',  # noqa: E501
+                asset=token_out,
+            )])
+
+        if context.tx_log.topics[0] == TRANSFER_TOPIC:
```

Comment:
can't you do all this have a common exit point?

Define the common attributes of all topics for the arguments of `ActionItem`. And then do an if/elif/else populating the per topic attributes but instead of returning as you do here in every if, return only once at the very end